flang/lib/Optimizer/CodeGen/Target.cpp

```diff
@@ -145,7 +146,7 @@ struct TargetX86_64 : public GenericTarget<TargetX86_64> {
       // <2 x t>   vector of 2 eleTy
       marshal.emplace_back(fir::VectorType::get(2, eleTy), AT{});
     } else if (sem == &llvm::APFloat::IEEEdouble()) {
-      // { double, double }   struct of 2 double
+      // ( t, t )   tuple of 2 eleTy
```

Comment:
Oh, sorry about that. They're a bit misleading in the current form, I didn't even realize they matched the final type, thought they were just out of date or copy-paste artifacts. Maybe we could have something like: "use a type that will be translated to a struct of 2 double {double, double}". Or would that be too verbose? We could also just remove them altogether to avoid confusion.

Reply:
I think some form of documentation is required here. A reader should not have to resort to reverse engineering the output of clang, for example.

Reply:
I updated the comments (basically I reverted my changes and added a few more words). Please let me know if this looks ok and I can commit this fixup back upstream as well.

Ideally LLVM itself would have some documentation on its ABI requirements, but I'm not aware of any. I did find a long [discussion](https://lists.llvm.org/pipermail/llvm-dev/2020-June/142055.html) about it from last summer. Didn't dig too much into that though.

Reply:
Agreed on your larger point. Thanks for updating the comments. Looks good.